be/src/exec/pipeline/exchange/local_exchange_source_operator.cpp

```diff
@@ -136,6 +136,19 @@ StatusOr<ChunkPtr> LocalExchangeSourceOperator::pull_chunk(RuntimeState* state)
     return std::move(chunk);
 }
 
+const size_t min_local_memory_limit = 1LL * 1024 * 1024;
+
+void LocalExchangeSourceOperator::enter_release_memory_mode() {
+    // limit the memory limit to a very small value so that the upstream will not write new data until all the data has been pushed to the downstream operators
+    _local_memory_limit = min_local_memory_limit;
+    size_t max_memory_usage = min_local_memory_limit * _memory_manager->get_max_input_dop();
+    _memory_manager->update_max_memory_usage(max_memory_usage);
```

Comment:
Should also modify `_local_buffer_almost_full()`?
```diff
    bool _local_buffer_almost_full() const {
-        return _local_memory_usage >= _memory_manager->get_memory_limit_per_driver() * 0.8;
+        return _local_memory_usage >= _local_memory_limit;
    }
```

Reply:
> Should also modify `_local_buffer_almost_full()`?
> 
> ```diff
>     bool _local_buffer_almost_full() const {
> -        return _local_memory_usage >= _memory_manager->get_memory_limit_per_driver() * 0.8;
> +        return _local_memory_usage >= _local_memory_limit;
>     }
> ```

fixed